### PR TITLE
Add support for executemany()

### LINKED ISSUE #807: `executemany()` does not translate pyformat dict params
### Summary

`Cursor.executemany()` with pyformat `%(name)s` SQL and a list of dicts raises a `SQLParseException` in crate-python 2.2.0. The pyformat→positional translation introduced in dev is never applied on the bulk code path, raw `%(name)s` SQL is forwarded to CrateDB's HTTP API, which does not understand pyformat syntax.

### Root cause

`executemany()` **does not** passes `seq_of_parameters` as `parameters`:

```python
# src/crate/client/cursor.py, line 113–121
def executemany(self, sql, seq_of_parameters):
    ...
    self.execute(sql, bulk_parameters=seq_of_parameters)  # ← bulk path
```

`execute()` only calls `_convert_named_to_positional()` when `isinstance(parameters, dict)` (line 101). The `bulk_parameters` argument is never inspected for pyformat translation.

As a result, the following JSON body is sent to CrateDB's `/_sql` endpoint:

```json
{
  "stmt": "INSERT INTO t (name, age) VALUES (%(name)s, %(age)s)",
  "bulk_args": [{"name": "Arthur", "age": 42}]
}
```

CrateDB's SQL parser rejects the `%(...)s` syntax:

```
SQLParseException[Failed to parse source [...]]
```

### Reproduction

```python
import crate.client as crate_client

conn = crate_client.connect("localhost:4200")
cur = conn.cursor()
cur.execute(
    "CREATE TABLE IF NOT EXISTS t (name TEXT PRIMARY KEY, age INT) "
    "WITH (number_of_replicas = 0)"
)

# FAILS
cur.executemany(
    "INSERT INTO t (name, age) VALUES (%(name)s, %(age)s)",
    [
        {"name": "Arthur", "age": 42},
        {"name": "Trillian", "age": 35},
    ],
)
```

### Related Issues: 
- Issue #806 
- PR #795

### PR DESCRIPTION
**Pull request by bgunebakan**
## Checklist

 - [x] Link to issue this PR refers to (if applicable): Fixes #807

### CODE REVIEW

**Review by kneth — APPROVED** [review]
LGTM

File: src/crate/client/cursor.py [hunk 1]
```diff
@@ -118,7 +149,16 @@ def executemany(self, sql, seq_of_parameters):
         """
         row_counts = []
         durations = []
-        self.execute(sql, bulk_parameters=seq_of_parameters)
+        bulk_parameters = seq_of_parameters
+        if (
+            bulk_parameters
+            and isinstance(bulk_parameters[0], dict)
+            and _NAMED_PARAM_RE.search(sql)
```
Comment (mfussenegger) [inline src/crate/client/cursor.py:161]: What happens if there is a mix of dict/non-dict args?

File: tests/client/test_cursor.py [hunk 2]
```diff
@@ -125,6 +125,53 @@ def test_cursor_executemany(mocked_connection):
...
+        "cols": [],
+        "duration": 123,
+        "results": [{"rowcount": 1}, {"rowcount": 1}],
+    }
+    with mock.patch.object(
+        mocked_connection.client, "sql", return_value=response
+    ):
+        cursor = mocked_connection.cursor()
+        cursor.executemany(
+            "INSERT INTO characters (name, age) VALUES (%(name)s, %(age)s)",
```
Comment (mfussenegger) [inline tests/client/test_cursor.py:145]: Can you also add a case where one named argument is used multiple times and check that it gets the same position number?

File: src/crate/client/cursor.py (hunk 1 above)
Comment (bgunebakan) [reply src/crate/client/cursor.py:161]: currently we raise an error. Do we have case to use mix of dict/non-dict together? I couldn't find how server handle that. What it should do?

Comment (mfussenegger) [reply src/crate/client/cursor.py:161]: Raising an error sounds good if it's one that makes it clear that mixing dict/non-dict together isn't supported.

Comment (bgunebakan) [reply src/crate/client/cursor.py:161]: I added edge case [here cursor.py:92](https://github.com/crate/crate-python/pull/810/changes#diff-fd97e585850a385a9eca471196fdfc76201e7cb13ac02fd3cd507d249d80d2f5R92) so it will raise `ProgrammingError` and added test for that.

File: tests/client/test_cursor.py (hunk 2 above)
Comment (bgunebakan) [reply tests/client/test_cursor.py:145]: Added tests https://github.com/crate/crate-python/pull/810/changes/10faeae4b4558652c9d8f83e3ef7c03f088ba226

**Review by mfussenegger — APPROVED** [review]
